### PR TITLE
multiple bug fixes

### PR DESCRIPTION
**Pull request by Sunil1993**
Added range filter
Fixed text filter
Default page number is changed to 0
@anantheshadiga

### CODE REVIEW

File: src/main/java/com/unbxd/client/search/SearchClient.java [hunk 1]
```diff
@@ -133,8 +134,40 @@ public SearchClient browse(List<String> nodeIds, Map<String, String> queryParams
      * @param values
      * @return this
      */
-    public SearchClient addFilter(String fieldName, String... values){
-        this.filters.put(fieldName, Arrays.asList(values));
+    public SearchClient addTextFilter(String fieldName, String... values){
+        if(textFilters.get(fieldName)==null) {
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/SearchClient.java]: Use containsKey

File: src/main/java/com/unbxd/client/search/SearchClient.java [hunk 2]
```diff
@@ -133,8 +134,40 @@ public SearchClient browse(List<String> nodeIds, Map<String, String> queryParams
...
+            for(String c : Arrays.asList(values)){
+                curr.add(c);
+            }
+            this.textFilters.put(fieldName,curr);
+        }
+        return this;
+    }
+
+    public SearchClient addRangeFilter(String fieldName, String start, String end){
+        if(rangeFilters.get(fieldName)==null){
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/SearchClient.java]: use containsKey

File: src/main/java/com/unbxd/client/search/SearchClient.java [hunk 3]
```diff
@@ -133,8 +134,40 @@ public SearchClient browse(List<String> nodeIds, Map<String, String> queryParams
...
-        this.filters.put(fieldName, Arrays.asList(values));
+    public SearchClient addTextFilter(String fieldName, String... values){
+        if(textFilters.get(fieldName)==null) {
+            this.textFilters.put(fieldName, Arrays.asList(values));
+        }else{
+            List<String> pre = this.textFilters.get(fieldName);
+            ArrayList<String> curr =new ArrayList<String>();
+            for(String c : pre){
+                curr.add(c);
+            }
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/SearchClient.java]: No need of this loop

File: src/main/java/com/unbxd/client/search/SearchClient.java [hunk 4]
```diff
@@ -201,14 +238,39 @@ private String generateUrl() throws SearchException {
...
+                                sb.append(URLEncoder.encode(key, __encoding) + ":[" + URLEncoder.encode(E.get(0) + " TO " + E.get(1), __encoding) + "]" + URLEncoder.encode(" OR ",__encoding));
+                            }
+                            size = size - 1;
+                        }
+                    }
+                    else{
+                            sb.append("&filter=" + URLEncoder.encode(key, __encoding) + ":[" + URLEncoder.encode(rangeFilters.get(key).get(0).get(0) + " TO " + rangeFilters.get(key).get(0).get(1), __encoding) + "]");
+                        }
+                    }
+                }
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/SearchClient.java]: Use StringUtils to join

File: src/test/java/com/unbxd/client/AutoSuggestClientTest.java [hunk 5]
```diff
@@ -32,12 +32,11 @@ public void test_autosuggest() throws AutoSuggestException, ConfigException {
...
 
-        System.out.println(response.toString());
         Assert.assertNotNull(response);
         Assert.assertEquals(0, response.getStatusCode());
-        Assert.assertNotEquals(0, response.getQueryTime());
+        Assert.assertNotNull(response.getQueryTime());
         Assert.assertEquals(0, response.getErrorCode());
         Assert.assertEquals("OK", response.getMessage());
-        Assert.assertNotEquals(0, response.getTotalResultsCount());
+        Assert.assertEquals(1925, response.getTotalResultsCount());
```
Comment (anantheshadiga) [inline src/test/java/com/unbxd/client/AutoSuggestClientTest.java]: Use assertNotEquals 0

File: src/main/java/com/unbxd/client/search/response/SearchResponse.java [hunk 6]
```diff
@@ -132,9 +138,14 @@ public Stats getStats(){
         return this._spellCorrections;
     }
 
+    public Banners get_banner() {
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/SearchResponse.java]: All the method name and variable name should follow camelcase

File: src/main/java/com/unbxd/client/search/response/Banner.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.Map;
+
+/**
+ * Created by unbxd on 8/19/15.
+ */
+public class Banner {
+    protected String algo;
+    protected List<String> imageUrl;
+    protected  List<String> landingUrl;
+    protected Map<String ,ArrayList> banner_names;
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/Banner.java]: you dont need this variable

File: src/main/java/com/unbxd/client/search/response/Banner.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Created by unbxd on 8/19/15.
+ */
+public class Banner {
+    protected String algo;
+    protected List<String> imageUrl;
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/Banner.java]: We dont need list here

File: src/main/java/com/unbxd/client/search/response/Banner.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+    protected  List<String> landingUrl;
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/Banner.java]: Same goes here

File: src/main/java/com/unbxd/client/search/response/SearchResponse.java (hunk 6 above)
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/SearchResponse.java]: This method should return List<Banner>

File: src/main/java/com/unbxd/client/search/SearchClient.java [hunk 10]
```diff
@@ -133,9 +136,35 @@ public SearchClient browse(List<String> nodeIds, Map<String, String> queryParams
      * @param values
      * @return this
      */
-    public SearchClient addFilter(String fieldName, String... values){
-        this.filters.put(fieldName, Arrays.asList(values));
+    public SearchClient addTextFilter(String fieldName, String... values){
+        if(textFilters.containsKey(fieldName)==false) {
+            this.textFilters.put(fieldName, Arrays.asList(values));
+        }else{
+            ArrayList<String> pre = new ArrayList<String>(this.textFilters.get(fieldName));
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/SearchClient.java]: Change return type to List

File: src/main/java/com/unbxd/client/search/SearchClient.java [hunk 11]
```diff
@@ -133,9 +136,35 @@ public SearchClient browse(List<String> nodeIds, Map<String, String> queryParams
...
+            for(String c : Arrays.asList(values)){
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/SearchClient.java]: Interate with the Array

File: src/main/java/com/unbxd/client/search/response/Banner.java [hunk 12]
```diff
@@ -0,0 +1,35 @@
...
+ * Created by unbxd on 8/19/15.
+ */
+public class Banner {
+    protected ArrayList<String> categories;
+    protected ArrayList<LinkedHashMap<String, String>> banners;
+    protected List<Banners> _banner;
+
+    public Banner(Map<String, Object> params) {
+        this._banner = new ArrayList<Banners>();
+        this.banners = (ArrayList<LinkedHashMap<String, String>>) params.get("banners");
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/Banner.java]: Validation

File: src/main/java/com/unbxd/client/search/response/Banners.java [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.LinkedHashMap;
+
+/**
+ * Created by unbxd on 8/20/15.
+ */
+public class Banners {
+    public String imageUrl;
+    public String  landingPageUrl;
+    public Banners(LinkedHashMap<String, String> banner) {
+        this.imageUrl = banner.get("imageUrl");
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/Banners.java]: Validation and use static final string

File: src/main/java/com/unbxd/client/search/response/Facet.java [hunk 14]
```diff
@@ -15,11 +15,23 @@
...
+    protected Integer position;
+    protected String displayName;
     protected List<FacetEntry> _facetEntries;
 
     protected Facet(String facetName, Map<String, Object> params){
         this.name = facetName;
         this._type = (String) params.get("type");
+        if(params.containsKey("position")){
+            this.position = (Integer) params.get("position");
+        }else {
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/Facet.java]: else is not required

File: src/main/java/com/unbxd/client/search/response/SearchResponse.java [hunk 15]
```diff
@@ -54,6 +56,11 @@ public SearchResponse(Map<String, Object> params){
                 this._facets = new Facets(facets);
             }
 
+            if(params.containsKey("banner")){
+                Map<String, Object> banners = (Map<String, Object>) params.get("banner");
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/SearchResponse.java]: Validation reqd

File: src/main/java/com/unbxd/client/search/response/SearchResponse.java [hunk 16]
```diff
@@ -133,8 +140,19 @@ public Stats getStats(){
     }
 
     /**
+      * @return Stats. Refer {@link Banner}
+     */
+    public List<Banners> getBanner() {
```
Comment (anantheshadiga) [inline src/main/java/com/unbxd/client/search/response/SearchResponse.java]: make it getBanners

File: src/test/java/com/unbxd/client/SearchClientTest.java [hunk 17]
```diff
@@ -35,8 +35,8 @@ public void test_search() throws SearchException, ConfigException {
         Assert.assertNotEquals(0, response.getQueryTime());
         Assert.assertEquals(0, response.getErrorCode());
         Assert.assertEquals("OK", response.getMessage());
-        Assert.assertNotEquals(0, response.getTotalResultsCount());
-        Assert.assertEquals(5, response.getResults().getResultsCount());
+        Assert.assertEquals(165, response.getTotalResultsCount());
```
Comment (anantheshadiga) [inline src/test/java/com/unbxd/client/SearchClientTest.java]: greater than 0

File: src/test/java/com/unbxd/client/SearchClientTest.java [hunk 18]
```diff
@@ -62,11 +65,12 @@ public void test_browse() throws SearchException, ConfigException {
         Assert.assertEquals(0, response.getErrorCode());
         Assert.assertEquals("OK", response.getMessage());
         Assert.assertNotEquals(0, response.getTotalResultsCount());
-        Assert.assertEquals(5, response.getResults().getResultsCount());
+        Assert.assertEquals(10, response.getResults().getResultsCount());
         Assert.assertEquals(1, response.getResults().getAt(0).getAttributes().size());
         Assert.assertNotNull(response.getResults().getAt(0).getAttributes().get("uniqueId"));
         Assert.assertNotNull(response.getStats());
```
Comment (anantheshadiga) [inline src/test/java/com/unbxd/client/SearchClientTest.java]: Does this test case pass? there is no request passing with parameter stats=price..

Comment (Sunil1993) [inline src/test/java/com/unbxd/client/SearchClientTest.java]: yeah we are passing stats in queryParams.


On Fri, Aug 21, 2015 at 1:16 AM, Ananthesh <notifications@github.com> wrote:

> In src/test/java/com/unbxd/client/SearchClientTest.java
> <https://github.com/unbxd/java-sdk/pull/2#discussion_r37572791>:
>
> > @@ -62,11 +65,12 @@ public void test_browse() throws SearchException, ConfigException {
> >          Assert.assertEquals(0, response.getErrorCode());
> >          Assert.assertEquals("OK", response.getMessage());
> >          Assert.assertNotEquals(0, response.getTotalResultsCount());
> > -        Assert.assertEquals(5, response.getResults().getResultsCount());
> > +        Assert.assertEquals(10, response.getResults().getResultsCount());
> >          Assert.assertEquals(1, response.getResults().getAt(0).getAttributes().size());
> >          Assert.assertNotNull(response.getResults().getAt(0).getAttributes().get("uniqueId"));
> >          Assert.assertNotNull(response.getStats());
>
> Does this test case pass? there is no request passing with parameter
> stats=price..
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/unbxd/java-sdk/pull/2/files#r37572791>.
>



-- 



Thanks,
Sunil

**Pull request merged**